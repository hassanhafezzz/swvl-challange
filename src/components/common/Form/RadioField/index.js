import React from 'react';
import { useField } from 'formik';

import classNames from 'classnames/bind';
import styles from './styles.module.css';

const cx = classNames.bind(styles);

const RadioField = ({ id, name, label, value, checked }) => {
  const [field, meta, helpers] = useField(name);
  const { setValue } = helpers;

  return (
    <div className={cx('wrapper')}>
      <input
        type="radio"
        id={id}
        name={name}
        onClick={() => {
          setValue(value);
        }}
        className={cx('radio')}
        checked={checked}
      />
      <label htmlFor={id} className={cx('label')}>
        {label}
      </label>
    </div>
  );
};

export default RadioField;
