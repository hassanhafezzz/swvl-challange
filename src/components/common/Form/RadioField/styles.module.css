.wrapper {
  margin: 10px 0;
  display: flex;
  align-items: center;
  justify-items: center;
}

.radio {
  cursor: pointer;
  margin-right: 10px;
  background-color: var(--color-dark-secondary);
}

.label {
  text-transform: capitalize;
  font-size: var(--small-paragraph);
  font-weight: var(--bold);
}
