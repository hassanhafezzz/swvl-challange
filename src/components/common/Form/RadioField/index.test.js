import React from 'react';
import { Formik } from 'formik';
import { render } from '@testing-library/react';

import RadioField from './index';

const component = (
  <Formik>
    <RadioField label="cash" name="paymentMethod" id="cash" value="cash" />
  </Formik>
);

test('input should have label', () => {
  const { getByLabelText } = render(component);
  const input = getByLabelText('cash');
  expect(input).toBeTruthy();
});

test('input snapshot', () => {
  const { container, asFragment } = render(component);
  expect(asFragment(container)).toMatchSnapshot();
});
