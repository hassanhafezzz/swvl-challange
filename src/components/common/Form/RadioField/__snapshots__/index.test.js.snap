// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <input
      class="radio"
      id="cash"
      name="paymentMethod"
      type="radio"
    />
    <label
      class="label"
      for="cash"
    >
      cash
    </label>
  </div>
</DocumentFragment>
`;
